plugins/locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/CompassManager.java

```diff
@@ -0,0 +1,74 @@
+package com.mapbox.mapboxsdk.plugins.locationlayer;
+
+import android.content.Context;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.SystemClock;
+
+/**
+ * This manager class handles compass events such as starting the tracking of device bearing, or when a new compass
+ * update occurs.
+ *
+ * @since 0.1.0
+ */
+class CompassManager implements SensorEventListener {
+
+  private final SensorManager sensorManager;
+  private CompassListener compassListener;
+
+  private Sensor rotationVectorSensor;
+  private float[] matrix = new float[9];
+  private float[] orientation = new float[3];
+
+  // CompassManager data
+  private long compassUpdateNextTimestamp = 0;
+
+  CompassManager(Context context, CompassListener compassListener) {
+    sensorManager = (SensorManager) context.getSystemService(Context.SENSOR_SERVICE);
+    rotationVectorSensor = sensorManager.getDefaultSensor(Sensor.TYPE_ROTATION_VECTOR);
+
+    this.compassListener = compassListener;
+  }
+
+  void onStart() {
```

Comment:
The `onStart` and `onStop` still need to be called from `LocationLayerPlugin` since an observer is only optional. If I added a lifecycle annotation here, it wouldn't resolve me having to call the onStart methods unless you had something else in mind and I misunderstood.